docs/runtime/runtime-metadata.md

```diff
@@ -0,0 +1,149 @@
+
+Before we dive into more detail, let's have a quick look at what information is included in the metadata. For demo purpose, we'll only show the metadata of [srml-sudo](https://substrate.dev/rustdocs/v1.0/srml_sudo/index.html) module in JSON format:
+
+```json
+{
+    "modules":[
+        {
+            "name":"sudo",
+            "prefix":"Sudo",
+            "storage":[
+                {
+                    "name":"Key",
+                    "modifier":"Default",
+                    "type":{
+                        "PlainType":"AccountId"
+                    },
+                    "fallback":"0x0000000000000000000000000000000000000000000000000000000000000000",
+                    "documentation":[
+                        " The `AccountId` of the sudo key."
+                    ]
+                }
+            ],
+            "calls":[
+                {
+                    "name":"sudo",
+                    "args":[
+                        {
+                            "name":"proposal",
+                            "type":"Proposal"
+                        }
+                    ],
+                    "documentation":[
+                        " Authenticates the sudo key and dispatches a function call with `Root` origin.",
+                        "",
+                        " The dispatch origin for this call must be _Signed_."
+                    ]
+                },
+                {
+                    "name":"set_key",
+                    "args":[
+                        {
+                            "name":"new",
+                            "type":"Address"
+                        }
+                    ],
+                    "documentation":[
+                        " Authenticates the current sudo key and sets the given AccountId (`new`) as the new sudo key.",
+                        "",
+                        " The dispatch origin for this call must be _Signed_."
+                    ]
+                }
+            ],
+            "events":[
+                {
+                    "name":"Sudid",
+                    "args":[
+                        "bool"
+                    ],
+                    "documentation":[
+                        " A sudo just took place."
+                    ]
+                },
+                {
+                    "name":"KeyChanged",
+                    "args":[
+                        "AccountId"
+                    ],
+                    "documentation":[
+                        " The sudoer just switched identity; the old key is supplied."
+                    ]
+                }
+            ]
+        }
+    ]
+}
+```
+
+As you can see, the metadata of overall runtime is an array of module's metadata. The included modules contain all the dependent SRML and custom runtime modules in your blockchain. 
+
+Now let's walk through each field in module's metadata:
```

Comment:
I'm not sure if this belongs here or in the reference docs. @shawntabrizi I don't see people coming here for guidance on their code. They are here to get an idea of what metadata is in the context of Substrate.

Things like "`name`: the function's name" don't add a lot of value. I wonder if we should compress this and have less description. Just like:

```
`storage`: All the public storage items. Includes:
    - `name`
    - `modifier`
    - etc...
```

Reply:
The `name` do feel verbose, but the description for other fields like `storage.type`, `storage.modifer`, `calls.args` could be helpful. I just make it this way for consistent.